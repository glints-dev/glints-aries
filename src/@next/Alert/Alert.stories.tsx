import { Meta, Story } from '@storybook/react';
import React, { useState } from 'react';

import { BaseContainer } from '../../Layout/GlintsContainer/GlintsContainer';
import { Button } from '../Button';
import { Typography } from '../Typography';
import { Alert, AlertProps } from './Alert';

(Alert as React.FunctionComponent<AlertProps>).displayName = 'Alert';

export default {
  title: '@next/Alert',
  component: Alert,
  decorators: [Story => <BaseContainer>{Story()}</BaseContainer>],
} as Meta;

const Template: Story<AlertProps> = args => {
  const [showAlert, setShowAlert] = useState(false);
  const triggerAlert = () => {
    setShowAlert(true);
  };

  return (
    <>
      <Button onClick={() => triggerAlert()}>Show Alert</Button>
      <Alert
        {...args}
        show={showAlert}
        onDismissed={() => setShowAlert(false)}
      />
    </>
  );
};

export const Interactive = Template.bind({});

Interactive.args = {
  title: 'Title',
  content: 'Alert content',
  status: 'info',
  children: (
    <Typography as="span" variant="body2" color="#017EB7">
      Learn more
    </Typography>
  ),
};
