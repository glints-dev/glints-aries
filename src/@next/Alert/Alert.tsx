import React, { useEffect } from 'react';
import { Icon } from '../Icon';
import { IconNames } from '../Icon/icons/icons';
import { Typography } from '../Typography';
import {
  StyledAlertContainer,
  StyledAlertContentColumn,
  StyledAlertCloseIconContainer,
} from './AlertStyle';

export type AlertProps = {
  title?: string;
  status?: 'success' | 'info' | 'warning' | 'error';
  content?: React.ReactNode;
  children?: React.ReactNode;
  show?: boolean;
  /** Duration of alert in miliseconds */
  duration?: number;
  onDismissed?: () => void;
};

const iconNameStatusMap: Record<AlertProps['status'], IconNames> = {
  success: 'ri-checkbox-circle-fill',
  info: 'ri-information-fill',
  warning: 'ri-error-warning-fill',
  error: 'ri-error-warning-fill',
};

export const Alert = ({
  show,
  title,
  content,
  status,
  duration = 4000,
  children,
  onDismissed,
}: AlertProps) => {
  useEffect(() => {
    if (!show) {
      return;
    }
    const timeId = setTimeout(() => {
      // onDismissed?.();
    }, duration);

    return () => {
      clearTimeout(timeId);
    };
  }, [duration, onDismissed, show]);

  const hasTitle = !!title;
  const iconStatus = iconNameStatusMap[status];
  if (status && !iconStatus) {
    console.warn(
      `Status "${status}" is not a valid Alert status, "info" will be used instead`
    );
  }
  const iconName = iconStatus || iconNameStatusMap['info'];

  if (!show) {
    return null;
  }

  return (
    <StyledAlertContainer data-titled={hasTitle} data-status={status || 'info'}>
      <div>
        <Icon name={iconName} />
      </div>
      <div>
        {hasTitle && (
          <StyledAlertContentColumn>
            <Typography as="div" variant="subtitle1">
              {title}
            </Typography>
          </StyledAlertContentColumn>
        )}
        <div>
          <StyledAlertContentColumn>
            <Typography as="div" variant="body1">
              {content}
            </Typography>
          </StyledAlertContentColumn>
        </div>
        <div>
          <StyledAlertContentColumn>
            <Typography as="div" variant="body1">
              {children}
            </Typography>
          </StyledAlertContentColumn>
        </div>
      </div>
      <StyledAlertCloseIconContainer onClick={() => onDismissed?.()}>
        <Icon name="ri-close" />
      </StyledAlertCloseIconContainer>
    </StyledAlertContainer>
  );
};
