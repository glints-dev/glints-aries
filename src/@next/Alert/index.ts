export * from './Alert';
export * from './AlertProvider';
export * from './AlertWithProvider';
export * from './AlertContext';
export * from './useAlert';
