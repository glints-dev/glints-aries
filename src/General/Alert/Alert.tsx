import * as React from 'react';

import classNames from 'classnames';

import {
  CheckmarkSolidIcon,
  WarningCircleSolidIcon,
  InfoSolidIcon,
  CloseIcon,
} from '../Icon/components';
import {
  AlertClose,
  AlertContainer,
  AlertIcon,
  AlertMessage,
} from './AlertStyle';

import { Greyscale, PrimaryColor, SecondaryColor } from '../../Utils/Colors';
import { isUndefined } from 'lodash';
import { DelayedUnmount } from '../../Utils/DelayedUnmount';

export const Alert = ({
  isOpen: externalIsVisible,
  autoClose,
  onClose: externalOnClose,
  type = 'info',
  className,
  message,
}: Props) => {
  const [internalIsVisible, setInternalIsVisible] = React.useState<boolean>(
    true
  );

  const isVisible = isUndefined(externalIsVisible)
    ? internalIsVisible
    : externalIsVisible;

  const internalOnClose = () => setInternalIsVisible(false);
  const onClose = isUndefined(externalOnClose)
    ? internalOnClose
    : externalOnClose;

  React.useEffect(() => {
    if (isVisible && autoClose) {
      const timeout = setTimeout(onClose, autoClose);
      return () => clearTimeout(timeout);
    }
  }, [isVisible, autoClose, onClose]);

  const handleKeyDown = React.useCallback((onClose: () => void) => {
    const listener = (e: React.KeyboardEvent<HTMLDivElement>) => {
      if (e.keyCode === 13 || e.keyCode === 27) {
        onClose();
      }
    };

    return listener;
  }, []);

  const { AlertTypeIcon, alertColor } = React.useMemo(() => {
    switch (type) {
      case 'success':
        return {
          AlertTypeIcon: CheckmarkSolidIcon,
          alertColor: SecondaryColor.darkgreen,
        };
      case 'warning':
        return {
          AlertTypeIcon: WarningCircleSolidIcon,
          alertColor: SecondaryColor.orange,
        };
      case 'danger':
        return {
          AlertTypeIcon: WarningCircleSolidIcon,
          alertColor: PrimaryColor.glintsred,
        };
      case 'info':
      default:
        return {
          AlertTypeIcon: InfoSolidIcon,
          alertColor: SecondaryColor.actionblue,
        };
    }
  }, [type]);

  return (
    // The delay is set to 290ms because the fade-out animation takes 300ms.
    // Setting them to the same value creates a visual glitch.
    <DelayedUnmount mount={isVisible} delay={290}>
      <AlertContainer
        className={classNames('aries-alert', className)}
        data-type={type}
        role="alertdialog"
        aria-hidden={!isVisible}
        data-open={isVisible}
        aria-describedby="alert-message"
        tabIndex={0}
        onKeyDown={handleKeyDown(onClose)}
        ref={node => node && node.focus()}
      >
        <AlertIcon>
          <AlertTypeIcon color={alertColor} />
        </AlertIcon>
        <AlertMessage className="alert-message">{message}</AlertMessage>
        <AlertClose
          className="alert-close"
          role="button"
          aria-label="Press Escape or Enter button to close alert"
          title="Close alert"
          onClick={onClose}
        >
          <CloseIcon color={Greyscale.devilsgrey} />
        </AlertClose>
      </AlertContainer>
    </DelayedUnmount>
  );
};

export type AlertType = 'success' | 'warning' | 'danger' | 'info';

export interface Props {
  /** Set Alert's style. */
  type?: AlertType;
  /** Set message for Alert. */
  message: string;
  /** A function to close Alert. onClose will be triggered if the close icon is clicked or the Alert has ran its autoClose duration or ESC/Enter is pressed */
  onClose(): void;
  /** Set Alert to show. */
  isOpen?: boolean;
  /** Set classname for Alert. */
  className?: string;
  /** Set period for Alert to close by itself. The value is in millisecond. */
  autoClose?: number;
}

export default Alert;
