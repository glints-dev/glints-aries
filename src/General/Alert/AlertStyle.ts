import styled, { keyframes } from 'styled-components';

import { ScreenSize } from '../../Utils/StyleConfig';
import { Shadow } from '../../Utils/Shadow';

const fadeDown = keyframes`
  0% {
    transform: translateY(-100%);
    top: 0;
  }

  100% {
    transform: translateY(0%);
    top: 16px;
  }
`;

const fadeUp = keyframes`
  0% {
    transform: translateY(0%);
    top: 12px;
  }

  100% {
    transform: translateY(-100%);
    top: 0;
  }
`;

export const AlertContainer = styled.div`
  position: fixed;
  left: 0;
  right: 0;
  margin: 0 auto;
  top: 16px;
  animation: ${fadeDown} 0.3s ease-in-out;
  &[aria-hidden='true'] {
    top: 0;
    animation: ${fadeUp} 0.3s ease-in-out;
  }
  display: flex;
  align-items: center;
  box-shadow: ${Shadow.down4};
  padding: 12px;
  border-radius: 4px;
  z-index: 5000;
  min-width: 296px;
  max-width: ${ScreenSize.tablet}px;
  outline: none;
  line-height: 1.5;
  font-size: 14px;
  color: #454545;

  @media (max-width: ${ScreenSize.tablet}px) {
    max-width: 95%;
  }

  background: #c5ecfa;
  &[data-type='success'] {
    background: #c2e2da;
  }
  &[data-type='warning'] {
    background: #fde9cc;
  }
  &[data-type='danger'] {
    background: #facbcc;
  }
`;

export const AlertIcon = styled.div`
  margin-right: 8px;
  line-height: 1;
  svg {
    display: block;
  }
`;

export const AlertMessage = styled.div`
  margin-right: 8px;
  flex-grow: 1;
`;

export const AlertClose = styled.div`
  line-height: 1;
  cursor: pointer;
  font-size: 10px;
`;
