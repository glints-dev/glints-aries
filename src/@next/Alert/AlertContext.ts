import { createContext } from 'react';
import { ComponentWithProviderProps } from '../../types/componentWithProvider';
import { AlertProps } from './Alert';

export type ShowAlertProps = Omit<AlertProps, 'show'>;

export type AlertContextProps = AlertProps &
  ComponentWithProviderProps<ShowAlertProps>;
export const AlertContext = createContext<
  AlertContextProps | Record<string, never>
>({});
