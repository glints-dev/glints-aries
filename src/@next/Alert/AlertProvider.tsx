import React, { useState } from 'react';
import { AlertProps } from './Alert';
import { AlertContext, ShowAlertProps } from './AlertContext';

export const AlertProvider = ({ children }: { children: React.ReactNode }) => {
  const [state, setState] = useState<AlertProps>(undefined);

  const open = (props: ShowAlertProps) => setState({ ...props, show: true });

  const close = () => setState({ show: false });

  const alertContextValue = { ...state, open, close };

  return (
    <AlertContext.Provider value={alertContextValue}>
      {children}
    </AlertContext.Provider>
  );
};
