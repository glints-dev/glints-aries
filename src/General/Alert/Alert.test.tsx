import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, fireEvent, wait, act } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { PrimaryColor, SecondaryColor } from '../../Utils/Colors';
import Alert, { AlertType } from './Alert';

describe('<Alert/> should match snapshot, when isOpen is falsy', () => {
  test('snapshot should be null regardless of alert type', () => {
    const snapshot = renderer
      .create(
        <Alert
          type={'success'}
          message="You have successfully applied to Glints as Software Engineer."
          isOpen={false}
          onClose={jest.fn()}
        />
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});

describe('<Alert/> should match snapshot, when isOpen is truthy', () => {
  const AlertComponent = ({ type }: { type: AlertType }) => (
    <Alert
      type={type}
      message="You have successfully applied to Glints as Software Engineer."
      isOpen={true}
      onClose={jest.fn()}
    />
  );

  test('type is success', () => {
    const snapshot = renderer
      .create(<AlertComponent type="success" />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  test('type is warning', () => {
    const snapshot = renderer
      .create(<AlertComponent type="warning" />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  test('type is danger', () => {
    const snapshot = renderer.create(<AlertComponent type="danger" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  test('type is info', () => {
    const snapshot = renderer.create(<AlertComponent type="info" />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  test('type is not a predefined type', () => {
    const snapshot = renderer
      .create(<AlertComponent type={'xxx' as AlertType} />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});

describe('<Alert/> should render correct icon color', () => {
  const AlertComponent = ({ type }: { type: AlertType }) => (
    <Alert
      type={type}
      message="You have successfully applied to Glints as Software Engineer."
      isOpen={true}
      onClose={jest.fn()}
    />
  );

  test('type is success', () => {
    const { queryAllByTestId } = render(<AlertComponent type="success" />);
    const icon = queryAllByTestId('icon-svg')[0];
    expect(icon).toHaveAttribute('fill', SecondaryColor.darkgreen);
  });

  test('type is warning', () => {
    const { queryAllByTestId } = render(<AlertComponent type="warning" />);
    const icon = queryAllByTestId('icon-svg')[0];
    expect(icon).toHaveAttribute('fill', SecondaryColor.orange);
  });

  test('type is danger', () => {
    const { queryAllByTestId } = render(<AlertComponent type="danger" />);
    const icon = queryAllByTestId('icon-svg')[0];
    expect(icon).toHaveAttribute('fill', PrimaryColor.glintsred);
  });

  test('type is info', () => {
    const { queryAllByTestId } = render(<AlertComponent type="info" />);
    const icon = queryAllByTestId('icon-svg')[0];
    expect(icon).toHaveAttribute('fill', SecondaryColor.actionblue);
  });

  test('type is not a predefined type', () => {
    const { queryAllByTestId } = render(
      <AlertComponent type={'xxx' as AlertType} />
    );
    const icon = queryAllByTestId('icon-svg')[0];
    expect(icon).toHaveAttribute('fill', SecondaryColor.actionblue);
  });
});

describe('<Alert/> isOpen', () => {
  const AlertComponent = ({ isOpen }: { isOpen: boolean }) => (
    <Alert
      type={'success'}
      message="You have successfully applied to Glints as Software Engineer."
      isOpen={isOpen}
      onClose={jest.fn()}
    />
  );

  test('should not be rendered when when isOpen is falsy', () => {
    const { queryByRole } = render(<AlertComponent isOpen={false} />);
    const alert = queryByRole('alertdialog');
    expect(alert).not.toBeInTheDocument();
  });

  test('should be rendered when when isOpen is truthy', () => {
    const { queryByRole } = render(<AlertComponent isOpen={true} />);
    const alert = queryByRole('alertdialog');
    expect(alert).toBeInTheDocument();
  });
});

describe('<Alert/> onClose', () => {
  const AlertComponent = ({ onClose }: { onClose: () => void }) => (
    <Alert
      type={'success'}
      message="You have successfully applied to Glints as Software Engineer."
      isOpen={true}
      onClose={onClose}
    />
  );

  test('click close button should call onClose', async () => {
    const onClose = jest.fn();
    const { queryByRole } = render(<AlertComponent onClose={onClose} />);
    const closeButton = queryByRole('button');
    fireEvent.click(closeButton);
    expect(onClose).toHaveBeenCalled();
  });

  test('press enter key should call onClose', async () => {
    const onClose = jest.fn();
    const { queryByRole } = render(<AlertComponent onClose={onClose} />);
    const alert = queryByRole('alertdialog');
    fireEvent.keyDown(alert, { key: 'Enter', keyCode: 13 });
    expect(onClose).toHaveBeenCalled();
  });

  test('press any key except enter and esc should not call onClose', async () => {
    const onClose = jest.fn();
    const { queryByRole } = render(<AlertComponent onClose={onClose} />);
    const alert = queryByRole('alertdialog');
    fireEvent.keyDown(alert, { key: 'a', keyCode: 65 });
    expect(onClose).not.toHaveBeenCalled();
  });

  test('press esc key should call onClose', async () => {
    const onClose = jest.fn();
    const { queryByRole } = render(<AlertComponent onClose={onClose} />);
    const alert = queryByRole('alertdialog');
    fireEvent.keyDown(alert, { key: 'Esc', keyCode: 27 });
    expect(onClose).toHaveBeenCalled();
  });
});

describe('<Alert/> autoClose', () => {
  test('onClose should be called after isOpen changed from truthy to falsy', async () => {
    const AlertComponent = ({
      onClose,
      isOpen,
    }: {
      onClose: () => void;
      isOpen: boolean;
    }) => (
      <Alert
        type={'success'}
        message="You have successfully applied to Glints as Software Engineer."
        isOpen={isOpen}
        onClose={onClose}
        autoClose={10}
      />
    );

    const onClose = jest.fn();
    const { rerender } = render(
      <AlertComponent onClose={onClose} isOpen={false} />
    );
    rerender(<AlertComponent onClose={onClose} isOpen={true} />);
    expect(onClose).not.toHaveBeenCalled();
    await wait(() => {
      // after autoClose timeout onClose should have been called
      expect(onClose).toHaveBeenCalled();
    });
  });
});

describe('<Alert/> timeout', () => {
  const AlertComponent = ({ isOpenProp }: { isOpenProp: boolean }) => {
    const [isOpen, setIsOpen] = React.useState(isOpenProp);
    return (
      <Alert
        type={'success'}
        message="You have successfully applied to Glints as Software Engineer."
        isOpen={isOpen}
        onClose={() => setIsOpen(false)}
      />
    );
  };

  test('click close button should close alert after timeout', async () => {
    const { queryByRole } = render(<AlertComponent isOpenProp={true} />);
    const alert = queryByRole('alertdialog');
    const closeButton = queryByRole('button');
    fireEvent.click(closeButton);
    expect(alert).toBeInTheDocument();
    await wait(() => {
      expect(alert).not.toBeInTheDocument();
    });
  });

  test('press enter key should close alert after timeout', async () => {
    const { queryByRole } = render(<AlertComponent isOpenProp={true} />);
    const alert = queryByRole('alertdialog');
    fireEvent.keyDown(alert, { key: 'Enter', keyCode: 13 });
    expect(alert).toBeInTheDocument();
    await wait(() => {
      expect(alert).not.toBeInTheDocument();
    });
  });

  test('press esc key should close alert after timeout', async () => {
    const { queryByRole } = render(<AlertComponent isOpenProp={true} />);
    const alert = queryByRole('alertdialog');
    fireEvent.keyDown(alert, { key: 'Esc', keyCode: 27 });
    expect(alert).toBeInTheDocument();
    await wait(() => {
      expect(alert).not.toBeInTheDocument();
    });
  });

  test('autoClose', async () => {
    let openAlert: () => void;
    const AlertComponent = ({ isOpenProp }: { isOpenProp: boolean }) => {
      const [isOpen, setIsOpen] = React.useState(isOpenProp);
      openAlert = () => setIsOpen(true);
      return (
        <Alert
          type={'success'}
          message="You have successfully applied to Glints as Software Engineer."
          isOpen={isOpen}
          onClose={() => setIsOpen(false)}
          autoClose={100}
        />
      );
    };

    const { queryByRole } = render(<AlertComponent isOpenProp={false} />);
    act(() => openAlert());
    const alert = queryByRole('alertdialog');
    expect(alert).toBeInTheDocument();
    await wait(() => {
      expect(alert).not.toBeInTheDocument();
    });
  });

  test('autoCloseTimeout should be cleared before component unmount', async () => {
    const useRefSpy = jest.spyOn(React, 'useRef');
    const clearTimeoutSpy = jest.spyOn(window, 'clearTimeout');
    const setTimeoutSpy = jest.spyOn(window, 'setTimeout');
    const { rerender, unmount } = render(
      <Alert
        type="success"
        message="You have successfully applied to Glints as Software Engineer."
        isOpen={false}
        onClose={jest.fn()}
        autoClose={100}
      />
    );

    rerender(
      <Alert
        type="success"
        message="You have successfully applied to Glints as Software Engineer."
        isOpen={true}
        onClose={jest.fn()}
        autoClose={100}
      />
    );

    unmount();
    expect(setTimeoutSpy.mock.calls.length).toEqual(
      // No idea what's calling the additional clearTimeout here. It's not us,
      // I checked.
      clearTimeoutSpy.mock.calls.length + 1
    );

    clearTimeoutSpy.mockRestore();
    useRefSpy.mockRestore();
  });
});

describe('<Alert/> focus', () => {
  const AlertComponent = ({ isOpen }: { isOpen: boolean }) => (
    <Alert
      type={'success'}
      message="You have successfully applied to Glints as Software Engineer."
      isOpen={isOpen}
      onClose={jest.fn()}
    />
  );

  test('alert dialog should have focus when isOpen changes from falsy to truthy', () => {
    const { queryByRole, rerender } = render(<AlertComponent isOpen={false} />);
    rerender(<AlertComponent isOpen={true} />);
    const alert = queryByRole('alertdialog');
    expect(alert).toHaveFocus();
  });
});
