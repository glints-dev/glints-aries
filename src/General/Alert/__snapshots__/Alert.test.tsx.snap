// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert/> should match snapshot, when isOpen is falsy snapshot should be null regardless of alert type 1`] = `null`;

exports[`<Alert/> should match snapshot, when isOpen is truthy type is danger 1`] = `
<div
  aria-describedby="alert-message"
  aria-hidden={false}
  className="AlertStyle__AlertContainer-fcjw72-0 hJPEX aries-alert"
  data-open={true}
  data-type="danger"
  onKeyDown={[Function]}
  role="alertdialog"
  tabIndex={0}
>
  <div
    className="AlertStyle__AlertIcon-fcjw72-1 iufUzP"
  >
    <svg
      className="IconStyle__VerticalCenteredSvg-x7d280-0 cJSNzi"
      data-testid="icon-svg"
      fill="#EC272B"
      height="1em"
      viewBox="0 0 100 100"
      width="1em"
    >
      <path
        d="M54.928 54.928V24.88h-9.856v30.048h9.856zm0 20.192V65.024h-9.856V75.12h9.856zM50 0c9.135 0 17.588 2.284 25.36 6.851 7.533 4.327 13.462 10.256 17.789 17.788C97.716 32.412 100 40.865 100 50c0 9.135-2.284 17.588-6.851 25.36-4.327 7.533-10.256 13.462-17.788 17.789C67.588 97.716 59.135 100 50 100c-9.135 0-17.588-2.284-25.36-6.851C17.106 88.742 11.177 82.772 6.85 75.24 2.284 67.468 0 59.055 0 50s2.284-17.468 6.851-25.24C11.258 17.228 17.228 11.258 24.76 6.85 32.532 2.284 40.945 0 50 0z"
      />
    </svg>
  </div>
  <div
    className="AlertStyle__AlertMessage-fcjw72-2 daCdpN alert-message"
  >
    You have successfully applied to Glints as Software Engineer.
  </div>
  <div
    aria-label="Press Escape or Enter button to close alert"
    className="AlertStyle__AlertClose-fcjw72-3 eUeiSL alert-close"
    onClick={[MockFunction]}
    role="button"
    title="Close alert"
  >
    <svg
      className="IconStyle__VerticalCenteredSvg-x7d280-0 cJSNzi"
      data-testid="icon-svg"
      fill="#666666"
      height="1em"
      viewBox="0 0 100 100"
      width="1em"
    >
      <path
        d="M100 10L90 0 50 40 10 0 0 10l40 40L0 90l10 10 40-40 40 40 10-10-40-40z"
      />
    </svg>
  </div>
</div>
`;

exports[`<Alert/> should match snapshot, when isOpen is truthy type is info 1`] = `
<div
  aria-describedby="alert-message"
  aria-hidden={false}
  className="AlertStyle__AlertContainer-fcjw72-0 hJPEX aries-alert"
  data-open={true}
  data-type="info"
  onKeyDown={[Function]}
  role="alertdialog"
  tabIndex={0}
>
  <div
    className="AlertStyle__AlertIcon-fcjw72-1 iufUzP"
  >
    <svg
      className="IconStyle__VerticalCenteredSvg-x7d280-0 cJSNzi"
      data-testid="icon-svg"
      fill="#017EB7"
      height="1em"
      viewBox="0 0 100 100"
      width="1em"
    >
      <path
        d="M99.997 49.33c.148 13.839-4.576 25.668-14.174 35.49-9.597 9.82-21.315 14.879-35.153 15.177-13.839.148-25.668-4.576-35.49-14.174C5.36 76.226.302 64.508.004 50.67-.145 36.83 4.58 25.002 14.177 15.18 23.774 5.36 35.492.302 49.33.004 63.17-.145 74.998 4.58 84.82 14.177c9.82 9.597 14.879 21.315 15.177 35.153zM55.134 19.868c-1.34-1.34-3.05-2.009-5.134-2.009-2.083 0-3.794.67-5.134 2.01-1.339 1.338-2.008 3.05-2.008 5.133s.67 3.794 2.008 5.133c1.34 1.34 3.05 2.01 5.134 2.01 2.083 0 3.794-.67 5.134-2.01 1.339-1.339 2.008-3.05 2.008-5.133s-.67-3.795-2.008-5.134zm4.687 61.157v-2.456h-4.91V40.18H40.178v2.679h4.91v35.711h-4.91v2.456h19.642z"
      />
    </svg>
  </div>
  <div
    className="AlertStyle__AlertMessage-fcjw72-2 daCdpN alert-message"
  >
    You have successfully applied to Glints as Software Engineer.
  </div>
  <div
    aria-label="Press Escape or Enter button to close alert"
    className="AlertStyle__AlertClose-fcjw72-3 eUeiSL alert-close"
    onClick={[MockFunction]}
    role="button"
    title="Close alert"
  >
    <svg
      className="IconStyle__VerticalCenteredSvg-x7d280-0 cJSNzi"
      data-testid="icon-svg"
      fill="#666666"
      height="1em"
      viewBox="0 0 100 100"
      width="1em"
    >
      <path
        d="M100 10L90 0 50 40 10 0 0 10l40 40L0 90l10 10 40-40 40 40 10-10-40-40z"
      />
    </svg>
  </div>
</div>
`;

exports[`<Alert/> should match snapshot, when isOpen is truthy type is not a predefined type 1`] = `
<div
  aria-describedby="alert-message"
  aria-hidden={false}
  className="AlertStyle__AlertContainer-fcjw72-0 hJPEX aries-alert"
  data-open={true}
  data-type="xxx"
  onKeyDown={[Function]}
  role="alertdialog"
  tabIndex={0}
>
  <div
    className="AlertStyle__AlertIcon-fcjw72-1 iufUzP"
  >
    <svg
      className="IconStyle__VerticalCenteredSvg-x7d280-0 cJSNzi"
      data-testid="icon-svg"
      fill="#017EB7"
      height="1em"
      viewBox="0 0 100 100"
      width="1em"
    >
      <path
        d="M99.997 49.33c.148 13.839-4.576 25.668-14.174 35.49-9.597 9.82-21.315 14.879-35.153 15.177-13.839.148-25.668-4.576-35.49-14.174C5.36 76.226.302 64.508.004 50.67-.145 36.83 4.58 25.002 14.177 15.18 23.774 5.36 35.492.302 49.33.004 63.17-.145 74.998 4.58 84.82 14.177c9.82 9.597 14.879 21.315 15.177 35.153zM55.134 19.868c-1.34-1.34-3.05-2.009-5.134-2.009-2.083 0-3.794.67-5.134 2.01-1.339 1.338-2.008 3.05-2.008 5.133s.67 3.794 2.008 5.133c1.34 1.34 3.05 2.01 5.134 2.01 2.083 0 3.794-.67 5.134-2.01 1.339-1.339 2.008-3.05 2.008-5.133s-.67-3.795-2.008-5.134zm4.687 61.157v-2.456h-4.91V40.18H40.178v2.679h4.91v35.711h-4.91v2.456h19.642z"
      />
    </svg>
  </div>
  <div
    className="AlertStyle__AlertMessage-fcjw72-2 daCdpN alert-message"
  >
    You have successfully applied to Glints as Software Engineer.
  </div>
  <div
    aria-label="Press Escape or Enter button to close alert"
    className="AlertStyle__AlertClose-fcjw72-3 eUeiSL alert-close"
    onClick={[MockFunction]}
    role="button"
    title="Close alert"
  >
    <svg
      className="IconStyle__VerticalCenteredSvg-x7d280-0 cJSNzi"
      data-testid="icon-svg"
      fill="#666666"
      height="1em"
      viewBox="0 0 100 100"
      width="1em"
    >
      <path
        d="M100 10L90 0 50 40 10 0 0 10l40 40L0 90l10 10 40-40 40 40 10-10-40-40z"
      />
    </svg>
  </div>
</div>
`;

exports[`<Alert/> should match snapshot, when isOpen is truthy type is success 1`] = `
<div
  aria-describedby="alert-message"
  aria-hidden={false}
  className="AlertStyle__AlertContainer-fcjw72-0 hJPEX aries-alert"
  data-open={true}
  data-type="success"
  onKeyDown={[Function]}
  role="alertdialog"
  tabIndex={0}
>
  <div
    className="AlertStyle__AlertIcon-fcjw72-1 iufUzP"
  >
    <svg
      className="IconStyle__VerticalCenteredSvg-x7d280-0 cJSNzi"
      data-testid="icon-svg"
      fill="#008464"
      height="1em"
      viewBox="0 0 100 100"
      width="1em"
    >
      <path
        d="M50 100C22.386 100 0 77.614 0 50S22.386 0 50 0s50 22.386 50 50-22.386 50-50 50zM38.823 63.644L23.206 48.038 18 53.329l20.823 20.809 44.878-44.846L78.406 24 38.823 63.644z"
      />
    </svg>
  </div>
  <div
    className="AlertStyle__AlertMessage-fcjw72-2 daCdpN alert-message"
  >
    You have successfully applied to Glints as Software Engineer.
  </div>
  <div
    aria-label="Press Escape or Enter button to close alert"
    className="AlertStyle__AlertClose-fcjw72-3 eUeiSL alert-close"
    onClick={[MockFunction]}
    role="button"
    title="Close alert"
  >
    <svg
      className="IconStyle__VerticalCenteredSvg-x7d280-0 cJSNzi"
      data-testid="icon-svg"
      fill="#666666"
      height="1em"
      viewBox="0 0 100 100"
      width="1em"
    >
      <path
        d="M100 10L90 0 50 40 10 0 0 10l40 40L0 90l10 10 40-40 40 40 10-10-40-40z"
      />
    </svg>
  </div>
</div>
`;

exports[`<Alert/> should match snapshot, when isOpen is truthy type is warning 1`] = `
<div
  aria-describedby="alert-message"
  aria-hidden={false}
  className="AlertStyle__AlertContainer-fcjw72-0 hJPEX aries-alert"
  data-open={true}
  data-type="warning"
  onKeyDown={[Function]}
  role="alertdialog"
  tabIndex={0}
>
  <div
    className="AlertStyle__AlertIcon-fcjw72-1 iufUzP"
  >
    <svg
      className="IconStyle__VerticalCenteredSvg-x7d280-0 cJSNzi"
      data-testid="icon-svg"
      fill="#F5A623"
      height="1em"
      viewBox="0 0 100 100"
      width="1em"
    >
      <path
        d="M54.928 54.928V24.88h-9.856v30.048h9.856zm0 20.192V65.024h-9.856V75.12h9.856zM50 0c9.135 0 17.588 2.284 25.36 6.851 7.533 4.327 13.462 10.256 17.789 17.788C97.716 32.412 100 40.865 100 50c0 9.135-2.284 17.588-6.851 25.36-4.327 7.533-10.256 13.462-17.788 17.789C67.588 97.716 59.135 100 50 100c-9.135 0-17.588-2.284-25.36-6.851C17.106 88.742 11.177 82.772 6.85 75.24 2.284 67.468 0 59.055 0 50s2.284-17.468 6.851-25.24C11.258 17.228 17.228 11.258 24.76 6.85 32.532 2.284 40.945 0 50 0z"
      />
    </svg>
  </div>
  <div
    className="AlertStyle__AlertMessage-fcjw72-2 daCdpN alert-message"
  >
    You have successfully applied to Glints as Software Engineer.
  </div>
  <div
    aria-label="Press Escape or Enter button to close alert"
    className="AlertStyle__AlertClose-fcjw72-3 eUeiSL alert-close"
    onClick={[MockFunction]}
    role="button"
    title="Close alert"
  >
    <svg
      className="IconStyle__VerticalCenteredSvg-x7d280-0 cJSNzi"
      data-testid="icon-svg"
      fill="#666666"
      height="1em"
      viewBox="0 0 100 100"
      width="1em"
    >
      <path
        d="M100 10L90 0 50 40 10 0 0 10l40 40L0 90l10 10 40-40 40 40 10-10-40-40z"
      />
    </svg>
  </div>
</div>
`;
